extract_model/utils.py

```diff
@@ -448,12 +448,30 @@ def preprocess_pom(ds):
 
     Parameters
     ----------
-    ds: xarray Dataset
+    ds : xr.Dataset
+        A dataset containing data described from POM output.
 
     Returns
     -------
-    Same Dataset but with some metadata added and/or altered.
+    xr.Dataset
+        Same Dataset but with some metadata added and/or altered.
     """
+    # The longitude and latitude variables are not recognized as valid coordinates
+    if "longitude" not in ds.cf.coords:
```

Comment:
extract_model is where I would do the generic:
```
ds = ds.cf.set_coords(["latitude", "longitude"])
```
which should work most or all of the time but we don't need to try too hard.